pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -209,6 +216,17 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 	}
 	managementClient := n.operatorKubeClient // We assume that operator is always running on the management cluster.
 
+	// If the work driver is not kube, we need to sync the work driver config secret
+	if workDriver != operatorapiv1.WorkDriverTypeKube {
```

Comment:
what happens if clustermanager is create at first and then secret is created?

Reply:
The operator will initially error, but subsequent reconcile will succeed after creating the secret.
Currently we don't watch the secret, do we need that?

Reply:
I think we need to reflect that the secret is missing.

Reply:
added condition to clustermanager CR's status to reflect that.